cli: TestZipIncludeAndExcludeFilesDataDriven failed

cli.TestZipIncludeAndExcludeFilesDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20967176?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20967176?buildTab=artifacts#/) on master @ [f51a5140106a9054bac864977ad23d3a2028bc50](https://github.com/cockroachdb/cockroach/commits/f51a5140106a9054bac864977ad23d3a2028bc50):


```
debug/system.migrations.txt
debug/system.namespace.txt
debug/system.privileges.txt
debug/system.protected_ts_meta.txt
debug/system.protected_ts_records.txt
debug/system.rangelog.txt
debug/system.replication_constraint_stats.txt
debug/system.replication_critical_localities.txt
debug/system.replication_stats.txt
debug/system.reports_meta.txt
debug/system.role_id_seq.txt
debug/system.role_members.txt
debug/system.role_options.txt
debug/system.scheduled_jobs.txt
debug/system.settings.txt
debug/system.span_configurations.txt
debug/system.sql_instances.txt
debug/system.sql_stats_cardinality.txt
debug/system.sqlliveness.txt
debug/system.statement_diagnostics.txt
debug/system.statement_diagnostics_requests.txt
debug/system.statement_hints.txt
debug/system.statement_statistics_limit_5000.txt
debug/system.table_statistics.txt
debug/system.table_statistics_locks.txt
debug/system.task_payloads.txt
debug/system.tenant_settings.txt
debug/system.tenant_tasks.txt
debug/system.tenant_usage.txt
debug/system.tenants.txt
debug/system.zones.txt
debug/tenant_ranges.err.txt
    zip_test.go:1571: -- test log scope end --
    zip_test.go:1509: 
          testdata/zip/file-filters/testzip_file_filters:2:
          --include-files=* --exclude-files=cpu.pprof
        output didn't match expected:
          @@ -67,10 +67,11 @@
           debug/nodes/1/crdb_internal.node_transaction_statistics.txt
           debug/nodes/1/crdb_internal.node_transactions.txt
           debug/nodes/1/crdb_internal.node_txn_execution_insights.txt
           debug/nodes/1/crdb_internal.node_txn_stats.txt
           debug/nodes/1/details.json
          +debug/nodes/1/goroutines/goroutine_dump.2026-01-13T16_56_32.465.double_since_last_dump.000001270.pb.gz
           debug/nodes/1/gossip.json
           debug/nodes/1/heap.pprof
           deterministic.log
           debug/nodes/1/lsm.txt
           debug/nodes/1/ranges.json
--- FAIL: TestZipIncludeAndExcludeFilesDataDriven/testzip_file_filters (41.96s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs @cockroachdb/server @cockroachdb/obs-india-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestZipIncludeAndExcludeFilesDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-58652

cli.TestZipIncludeAndExcludeFilesDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20984082?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20984082?buildTab=artifacts#/) on master @ [81f046c5eb268fcf87c8c34fa099dc291cdf24aa](https://github.com/cockroachdb/cockroach/commits/81f046c5eb268fcf87c8c34fa099dc291cdf24aa):


```
debug/system.migrations.txt
debug/system.namespace.txt
debug/system.privileges.txt
debug/system.protected_ts_meta.txt
debug/system.protected_ts_records.txt
debug/system.rangelog.txt
debug/system.replication_constraint_stats.txt
debug/system.replication_critical_localities.txt
debug/system.replication_stats.txt
debug/system.reports_meta.txt
debug/system.role_id_seq.txt
debug/system.role_members.txt
debug/system.role_options.txt
debug/system.scheduled_jobs.txt
debug/system.settings.txt
debug/system.span_configurations.txt
debug/system.sql_instances.txt
debug/system.sql_stats_cardinality.txt
debug/system.sqlliveness.txt
debug/system.statement_diagnostics.txt
debug/system.statement_diagnostics_requests.txt
debug/system.statement_hints.txt
debug/system.statement_statistics_limit_5000.txt
debug/system.table_statistics.txt
debug/system.table_statistics_locks.txt
debug/system.task_payloads.txt
debug/system.tenant_settings.txt
debug/system.tenant_tasks.txt
debug/system.tenant_usage.txt
debug/system.tenants.txt
debug/system.zones.txt
debug/tenant_ranges.err.txt
    zip_test.go:1571: -- test log scope end --
    zip_test.go:1509: 
          testdata/zip/file-filters/testzip_file_filters:2:
          --include-files=* --exclude-files=cpu.pprof
        output didn't match expected:
          @@ -67,10 +67,11 @@
           debug/nodes/1/crdb_internal.node_transaction_statistics.txt
           debug/nodes/1/crdb_internal.node_transactions.txt
           debug/nodes/1/crdb_internal.node_txn_execution_insights.txt
           debug/nodes/1/crdb_internal.node_txn_stats.txt
           debug/nodes/1/details.json
          +debug/nodes/1/goroutines/goroutine_dump.2026-01-15T23_56_45.630.double_since_last_dump.000001261.pb.gz
           debug/nodes/1/gossip.json
           debug/nodes/1/heap.pprof
           deterministic.log
           debug/nodes/1/lsm.txt
           debug/nodes/1/ranges.json
--- FAIL: TestZipIncludeAndExcludeFilesDataDriven/testzip_file_filters (22.03s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #161113 cli: TestZipIncludeAndExcludeFilesDataDriven failed [C-test-failure O-robot T-observability branch-release-26.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestZipIncludeAndExcludeFilesDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/cli/debug_list_files_test.go
+++ b/pkg/cli/debug_list_files_test.go
@@ -26,6 +26,8 @@ func TestExtractTimestamp(t *testing.T) {
 		{"goroutine_dump.double_since_last_dump.2021-04-23T09_18_23.1231", "2021-04-23 09:18"},
 		// v20.x and later goroutine dumps.
 		{"goroutine_dump.2021-03-11T08_13_57.498.double_since_last_dump.000001137.txt", "2021-03-11 08:13"},
+		// v26.1 and later goroutine dumps.
+		{"goroutine_dump.2021-03-11T08_13_57.498.double_since_last_dump.000001137.pb.gz", "2021-03-11 08:13"},
 		// v20.1 and later memstats.
 		{"memstats.2021-04-22T18_31_54.413.371441664.txt", "2021-04-22 18:31"},
 		// v1.0 heap profile names.

--- a/pkg/cli/zip_test.go
+++ b/pkg/cli/zip_test.go
@@ -1494,7 +1494,7 @@ func trimNonDeterministicZipOutputFiles(out string) string {
 	out = re.ReplaceAllString(out, ``)
 	re = regexp.MustCompile(`(?m).*(memprof|memstats|memmonitoring).*\.(txt|pprof)$` + "\n")
 	out = re.ReplaceAllString(out, ``)
-	re = regexp.MustCompile(`(?m).*goroutine_dump.*\.txt\.gz$` + "\n")
+	re = regexp.MustCompile(`(?m).*goroutine_dump.*\.pb\.gz$` + "\n")
 	out = re.ReplaceAllString(out, ``)
 	return out
 }

--- a/pkg/server/storage_api/files_test.go
+++ b/pkg/server/storage_api/files_test.go
@@ -94,16 +94,16 @@ func TestStatusGetFiles(t *testing.T) {
 	t.Run("goroutines", func(t *testing.T) {
 
 		// regex for goroutine file names manually added
-		reDump := regexp.MustCompile(`goroutine_dump\d+.txt.gz`)
+		reDump := regexp.MustCompile(`goroutine_dump\d+.pb.gz`)
 		// regex for goroutine file names dumped by goroutinedumper
-		reOOMDump := regexp.MustCompile("goroutine_dump.*.double_since_last_dump.*.txt.gz")
+		reOOMDump := regexp.MustCompile("goroutine_dump.*.double_since_last_dump.*.pb.gz")
 		// regex for content of goroutine files manually added
 		reDumpContent := regexp.MustCompile(`Goroutine dump \d+`)
 
 		const testFilesNo = 3
 		for i := 0; i < testFilesNo; i++ {
 			testGoroutineDir := filepath.Join(storeSpec.Path, "logs", base.GoroutineDumpDir)
-			testGoroutineFile := filepath.Join(testGoroutineDir, fmt.Sprintf("goroutine_dump%d.txt.gz", i))
+			testGoroutineFile := filepath.Join(testGoroutineDir, fmt.Sprintf("goroutine_dump%d.pb.gz", i))
 			if err := os.MkdirAll(testGoroutineDir, os.ModePerm); err != nil {
 				t.Fatal(err)
 			}

